lib/cocoapods-core/cdn_source.rb

```diff
@@ -0,0 +1,210 @@
+    #         the name of the Pod.
+    #
+    def versions(name)
+      return nil unless specs_dir
+      raise ArgumentError, 'No name' unless name
+
+      return @versions_by_name[name] unless @versions_by_name[name].nil?
+
+      versions_file_name = pod_path_partial(name).join('index.txt')
+      download_file(versions_file_name)
+
+      loaders = []
+      @versions_by_name[name] ||= local_file(versions_file_name) do |file|
+        file.map do |v|
+          basename = v.chomp
+          # Optimization: ensure all the podspec files at least exist. The correct one will get refreshed
+          # in #specification_path regardless.
+          unless local_file_exist?(pod_path_partial(name).join("#{basename}/#{name}.podspec.json"))
+            loaders << Concurrent::Promise.execute(:executor => @executor) do
+              download_file(pod_path_partial(name).join("#{basename}/#{name}.podspec.json"))
+            end
+          end
+          begin
+            Version.new(basename) if basename[0, 1] != '.'
+          rescue ArgumentError
+            raise Informative, 'An unexpected version directory ' \
+            "`#{basename}` was encountered for the " \
+            "`#{pod_dir}` Pod in the `#{name}` repository."
+          end
+        end
+      end.compact.sort.reverse
+      Concurrent::Promise.zip(*loaders).wait!
+      @versions_by_name[name]
+    end
+
+    # Returns the path of the specification with the given name and version.
+    #
+    # @param  [String] name
+    #         the name of the Pod.
+    #
+    # @param  [Version,String] version
+    #         the version for the specification.
+    #
+    # @return [Pathname] The path of the specification.
+    #
+    def specification_path(name, version)
+      raise ArgumentError, 'No name' unless name
+      raise ArgumentError, 'No version' unless version
+
+      relative_podspec = pod_path_partial(name).join("#{version}/#{name}.podspec.json")
+      download_file(relative_podspec)
+      pod_path(name).join("#{version}/#{name}.podspec.json")
+    end
+
+    # @!group Searching the source
+    #-------------------------------------------------------------------------#
+
+    # @return [Set] a set for a given dependency. The set is identified by the
+    #               name of the dependency and takes into account subspecs.
+    #
+    # @note   This method is optimized for fast lookups by name, i.e. it does
+    #         *not* require iterating through {#pod_sets}
+    #
+    # @todo   Rename to #load_set
+    #
+    def search(query)
+      unless specs_dir
+        raise Informative, "Unable to find a source named: `#{name}`"
+      end
+      if query.is_a?(Dependency)
+        query = query.root_name
+      end
+
+      found = download_file(pod_path_partial(query).join('index.txt'))
+
+      if found
+        set = set(query)
+        set if set.specification_name == query
+      end
+    end
+
+    def git?
+      false
+    end
+  end
```

Comment:
I think `update` will need to be overridden since the default implementation calls out to `git`

Reply:
I checked, and we don't actually get to CDNSource#update. The source manager says it's not a git repo, which is true. Luckily for us, there's no updating required for this repo type!